extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -773,78 +773,94 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
 		const addButtonComponent: DialogButton = {
 			buttonAriaLabel: localizedConstants.AddFilegroupText,
-			buttonHandler: () => this.onAddDatabaseFileGroupsButtonClicked(this.rowsFilegroupsTable)
+			buttonHandler: () => this.onAddDatabaseFileGroupsButtonClickedDeclarative(this.rowsFilegroupsTable)
 		};
 		const removeButtonComponent: DialogButton = {
 			buttonAriaLabel: localizedConstants.RemoveButton,
-			buttonHandler: () => this.onRemoveDatabaseFileGroupsButtonClicked(this.rowsFilegroupsTable)
+			buttonHandler: () => this.onRemoveDatabaseFileGroupsButtonClickedDeclarative(this.rowsFilegroupsTable)
 		};
-		this.rowsFileGroupButtonContainer = this.addButtonsForTable(this.rowsFilegroupsTable, addButtonComponent, removeButtonComponent);
+		this.rowsFileGroupButtonContainer = this.addButtonsForDeclarativeTable(this.rowsFilegroupsTable, addButtonComponent, removeButtonComponent);
 		this.disposables.push(
-			this.rowsFilegroupsTable.onCellAction(async (arg: azdata.ICheckboxCellActionEventArgs) => {
-				let filegroup = this.rowDataFileGroupsTableRows[arg.row];
-				// Read-Only column
-				if (arg.column === 2) {
-					filegroup.isReadOnly = arg.checked;
-				}
-				// Default column
-				if (arg.column === 3) {
-					this.updateFilegroupsDefaultColumnValues(arg, filegroup, FileGroupType.RowsFileGroup);
-				}
-				// Autogrow all files column
-				if (arg.column === 4) {
-					filegroup.autogrowAllFiles = arg.checked;
-				}
+			this.rowsFilegroupsTable.onDataChanged(async (changedData) => {
+				if (changedData.row > -1) {
+					let filegroup = this.rowDataFileGroupsTableRows[changedData.row];
+					// Read-Only column
+					if (changedData.column === 2) {
+						filegroup.isReadOnly = changedData.value;
+					}
+					// Default column
+					if (changedData.column === 3) {
+						this.updateFilegroupsDefaultColumnValuesDeclarative(changedData, filegroup, FileGroupType.RowsFileGroup);
+					}
+					// Autogrow all files column
+					if (changedData.column === 4) {
+						filegroup.autogrowAllFiles = changedData.value;
+					}
 
-				// Refresh the table with updated data
-				let data = this.getTableData(FileGroupType.RowsFileGroup);
-				await this.setTableData(this.rowsFilegroupsTable, data);
-				this.onFormFieldChange();
+					// Refresh the table with updated data
+					let data = this.getDeclarativeTableData(FileGroupType.RowsFileGroup);
+					await this.setDeclarativeTableData(this.rowsFilegroupsTable, data);
+					this.onFormFieldChange();
+				}
 			}),
-			this.rowsFilegroupsTable.onRowSelected(
-				async () => {
-					if (this.rowsFilegroupsTable.selectedRows.length === 1) {
-						const fileGroup = this.rowDataFileGroupsTableRows[this.rowsFilegroupsTable.selectedRows[0]];
-						this.rowsFilegroupNameContainer.display = fileGroup.id < 0 ? 'inline-flex' : 'none';
-						this.rowsFilegroupNameInput.value = fileGroup.name;
-						this.onFormFieldChange();
-					}
+			this.rowsFilegroupsTable.onRowSelected(async () => {
```

Comment:
this is the `async` that was supposed to be removed in the original commit, right?

Reply:
Yeah, removed it in the wrong place.